### ISSUE TITLE
Cloudflared server not starting

### ISSUE THREAD

**Issue by mnt4** [oobabooga/text-generation-webui#1524]
### Describe the bug

Adding --public-api to the flags causes a traceback instead of starting a cloudflared server. flask-clouflared is installed separately, despite not being included in requirements.txt

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Reproduction

1. Install flask-cloudflared
2. Run python server.py --share --model_type llama --no-stream --api --public-api --wbits 4 --groupsize 128
3. Get a Traceback error

### Logs

```shell
Traceback (most recent call last):
  File "/usr/lib/python3.9/threading.py", line 980, in _bootstrap_inner
    self.run()
  File "/usr/lib/python3.9/threading.py", line 917, in run
    self._target(*self._args, **self._kwargs)
  File "/content/drive/MyDrive/text-generation-webui/extensions/api/util.py", line 69, in _start_cloudflared
Exception in thread Thread-5:
Traceback (most recent call last):
  File "/usr/lib/python3.9/threading.py", line 980, in _bootstrap_inner
    raise Exception('Could not start cloudflared.')
Exception: Could not start cloudflared.
    self.run()
  File "/usr/lib/python3.9/threading.py", line 917, in run
    self._target(*self._args, **self._kwargs)
  File "/content/drive/MyDrive/text-generation-webui/extensions/api/util.py", line 69, in _start_cloudflared
    raise Exception('Could not start cloudflared.')
Exception: Could not start cloudflared.
```

### System Info

```shell
Google Colab
```

**Comment by TruthSearchers** [comment]
Same issue

**Comment by oobabooga** [comment]
The API extension hides the real error, which is

> OSError: [Errno 8] Exec format error: '/tmp/cloudflared-linux-amd64'

It looks like cloudfare messed up their reverse proxy executable. It should fix itself eventually

**Comment by mnt4** [comment]
I'm possibly being very wrong, but it seems that the actual `flask-clouflared` package is working well
![image](https://user-images.githubusercontent.com/58336655/234203564-7300ad8c-08b3-4792-82a5-a3ccebb34374.png)

**Comment by mnt4** [comment]
I made a fork of `flask_cloudflared` that fixes the issue temporarily
It can be used via `pip install git+https://github.com/mnt4/flask-cloudflared`

**Comment by TruthSearchers** [comment]
It worked fined now thanks bro 
But there it's giving instead of https it's giving wss at starting for streaming one 

Is it same for you?

**Comment by TruthSearchers** [comment]
It's working fine now bro thanks ☺️

**Comment by laughinggaschambers** [comment]
I'm getting the same issue
> I made a fork of `flask_cloudflared` that fixes the issue temporarily It can be used via `pip install git+https://github.com/mnt4/flask-cloudflared`

I'm using the one click install version, running the command doesn't fix it

**Comment by juliankolbe** [comment]
The real issue why this bug is happening is that both the blocking api and then streaming api start the _run_cloudflared more or less at the same time, and if this file '/tmp/cloudflared-linux-amd64' is not downloaded and ready fast enough, they collide as the file is still downloading. This is only a working theory, but on my host machine it had no problem, but as soon as this was running in docker it did. 

TLDR: I changed the extensions/api/script.py and added a time.sleep(5) in between calls, fixed it for me.

```def setup():
    blocking_api.start_server(BLOCKING_PORT, share=shared.args.public_api)
    time.sleep(5)
    streaming_api.start_server(STREAMING_PORT, share=shared.args.public_api)

**Comment by dje3392** [comment]
I've tried adding time.sleep and installing the flask cloudflared fix with no success :(

**Comment by pato-pan** [comment]
for some users, you simply have to give execute permissions to /tmp/cloudflared-linux-amd64

like this
`chmod +x /tmp/cloudflared-linux-amd64`

This should be clear from the logs

`PermissionError: [Errno 13] Permission denied: '/tmp/cloudflared-linux-amd64'`

but that's easy to miss when you are also facing other errors. In my case I didn't have cloudflared installed on my os (Installing it on conda won't work, it has to be on the os) and I got the permission denied error, so when I installed it and the logs looked different I did not think of checking that file again, I assumed installing cloudflare would fix it and moved on to the next error I was facing.

**Comment by ChenMorSays** [comment]
Still doesnt work for me, any updates?

**Comment by adityasingh8010** [comment]
> The real issue why this bug is happening is that both the blocking api and then streaming api start the _run_cloudflared more or less at the same time, and if this file '/tmp/cloudflared-linux-amd64' is not downloaded and ready fast enough, they collide as the file is still downloading. This is only a working theory, but on my host machine it had no problem, but as soon as this was running in docker it did.
> 
> TLDR: I changed the extensions/api/script.py and added a time.sleep(5) in between calls, fixed it for me.
> 
> ```
>     blocking_api.start_server(BLOCKING_PORT, share=shared.args.public_api)
>     time.sleep(5)
>     streaming_api.start_server(STREAMING_PORT, share=shared.args.public_api)
> ```

This hypothesis might be correct because sometimes it would run for me and sometimes it would crash. But adding in the 5 second sleep fixed the issue for me

**Comment by kushshrivastava** [comment]
I managed to run the API on one colab notebook and use it in another colab notebook using these steps

**In first colab notebook**
Clone text-generation-webui and install all requirements
```
!pip install flask-cloudflared
%cd /content/text-generation-webui
!python3.10 server.py --loader ExLlama --model /content/text-generation-webui/models/Llama-2-13b-Chat-GPTQ --public-api

```
this will return a url something like this
Starting non-streaming server at public url https://ABC-flag.trycloudflare.com/api

**In second colab notebook**
Go to api-examples/api-example.py
replace the above url in place of HOST
```
HOST = 'ABC-flag.trycloudflare.com'
URI = f'https://{HOST}/api/v1/generate'
```
Then run the file
`!python3.10 api-example.py`
